src/main/java/database/StorageDealerImpl.java

```diff
@@ -0,0 +1,21 @@
+package database;
+
+import java.util.Map;
+
+public class StorageDealerImpl implements StorageDealer {
+    @Override
+    public void checkBalance(Map<String, Integer> transactions) {
+        transactions.entrySet().stream()
+                .filter(entry -> entry.getValue() < 0)
+                .findAny()
+                .ifPresent(entry -> {
+                    throw new RuntimeException("Negative balance for fruit: " + entry.getKey()
+                            + " with quantity: " + entry.getValue());
+                });
+    }
+
+    @Override
+    public void updateDatabase(Map<String, Integer> transactions) {
+        Storage.updateStorage(transactions);
```

Comment:
It's a code quality issue. It's better to pass a copy of the `transactions` map to the `updateStorage` method, to prevent it from being accidentally modified elsewhere in the code. You can create a copy of the map using `new HashMap<>(transactions)`.